src/ucp/wireup/select.c

```diff
@@ -930,30 +904,34 @@ static ucs_status_t ucp_wireup_add_am_lane(ucp_ep_h ep, const ucp_ep_params_t *p
                                   UCT_IFACE_FLAG_CB_SYNC;
     criteria.local_iface_flags  = UCT_IFACE_FLAG_AM_BCOPY;
     criteria.calc_score         = ucp_wireup_am_score_func;
-    ucp_wireup_fill_ep_params_criteria(&criteria, params);
+    ucp_wireup_fill_ep_params_criteria(&criteria, select_ctx->params);
 
-    if (ucs_test_all_flags(ucp_ep_get_context_features(ep), UCP_FEATURE_TAG |
-                                                            UCP_FEATURE_WAKEUP)) {
+    if (ucs_test_all_flags(ucp_ep_get_context_features(ep),
+                           UCP_FEATURE_TAG | UCP_FEATURE_WAKEUP)) {
         criteria.local_iface_flags |= UCP_WORKER_UCT_UNSIG_EVENT_CAP_FLAGS;
     }
 
-    status = ucp_wireup_select_transport(ep, address_list, address_count, &criteria,
-                                         -1, -1, -1, -1, 1, am_info);
+    status = ucp_wireup_select_transport(ep, select_ctx->address_list,
+                                         select_ctx->address_count, &criteria,
+                                         -1, -1, -1, -1, 1, &select_ctx->am_info);
     if (status != UCS_OK) {
         return status;
     }
 
+    am_info      = &select_ctx->am_info;
+    dst_md_index = select_ctx->address_list[am_info->addr_index].md_index;
+
     /* If the remote side is not p2p and has only signaled-am wakeup, it may
      * deactivate its interface and wait for signaled active message to wake up.
      * Use a proxy lane which would send the first active message as signaled to
-     * make sure the remote interface will indeed wake up.
-     */
-    is_proxy = ucp_wireup_is_lane_proxy(ep, am_info->rsc_index,
-                                        address_list[am_info->addr_index].iface_attr.cap_flags);
+     * make sure the remote interface will indeed wake up. */
+    remote_cap_flags = select_ctx->address_list[am_info->addr_index].iface_attr.cap_flags;
+    is_proxy         = ucp_wireup_is_lane_proxy(ep, am_info->rsc_index,
+                                                remote_cap_flags);
 
-    ucp_wireup_add_lane_desc(lane_descs, num_lanes_p, am_info,
-                             address_list[am_info->addr_index].md_index,
-                             UCP_WIREUP_LANE_USAGE_AM, is_proxy);
+    ucp_wireup_add_lane_desc(select_ctx->lane_descs, &select_ctx->num_lanes,
```

Comment:
maybe pass select_ctx to ucp_wireup_add_lane_desc?

Reply:
done